subsys/pm/device.c

```diff
@@ -207,6 +207,58 @@ void pm_device_children_action_run(const struct device *dev,
 }
 #endif /* CONFIG_PM_DEVICE_POWER_DOMAIN */
 
+#if defined(CONFIG_PM_DEVICE_POWER_DOMAIN)
+void pm_device_domain_children_turn_on(const struct device *dev)
+{
+	struct pm_device_base *pm_base = dev->pm_base;
+	const struct device *child;
+
+	for (uint16_t i = 0; i < pm_base->domain_children_size; i++) {
+		child = pm_base->domain_children[i];
+
+		if (child == NULL) {
+			continue;
+		}
+
+		if (pm_device_action_run(child, PM_DEVICE_ACTION_TURN_ON) == 0) {
+			continue;
+		}
+
+		LOG_ERR("%s could not be %s", child->name, "turned on");
+	}
+}
+
+void pm_device_domain_children_turn_off(const struct device *dev)
+{
+	struct pm_device_base *pm_base = dev->pm_base;
+	const struct device *child;
+
+	for (uint16_t i = 0; i < pm_base->domain_children_size; i++) {
+		child = pm_base->domain_children[i];
+
+		if (child == NULL) {
+			continue;
+		}
+
+		if (pm_device_action_run(child, PM_DEVICE_ACTION_TURN_OFF) == 0) {
+			continue;
+		}
+
+		LOG_ERR("%s could not be %s", child->name, "turned off");
+	}
+}
```

Comment:
We should have only one function that receives the action to take. These two functions are doing pretty much the same thing with different actions. The main reason for it is to reduce ROM / flash usage.

Reply:
optimized to call single onoff function

Reply:
https://github.com/zephyrproject-rtos/zephyr/blob/e96bf44d934050b404e53aefdcb9756290b4c18e/tests/lib/devicetree/api/src/main.c#L377